### PR TITLE
Exams

### PR DESCRIPTION
**Pull request by GhostDolphin**

### CODE REVIEW

File: Tasks/ip.js [hunk 1]
```diff
@@ -1,17 +1,3 @@
...
-    for ( const b of B) {
-  a[j] = parseInt(b);
-  j++;
-  }};
-  return a;
-};
-
-const result = Parseip('127.0.0.1')
-console.log(result)
+const parseIp = ip => ip === '' ? 0 : ip.split('.').map(e => +e);
```
Comment (tshemsedinov) [inline Tasks/ip.js:3]: - Use parseInt is preferable here
- Intermediate variables will make this code more readable
- Use round brackets even for single arguments lambda functions
- Prefer if-statement here for readability
- Improve naming, `e` is not ok

File: Tasks/plane.js [hunk 2]
```diff
@@ -1,19 +1,8 @@
...
-    arr[i] = res[j - 1]
-    res.push(value);
-    };
-  };
-  return res
-}
-
-const result = const_plane([1, [[2]], [3, 4], [5], [6, [7, 8]]]);
-console.log(result);
+const plane = arr => {
```
Comment (tshemsedinov) [inline Tasks/plane.js:3]: Use round brackets even for single argument lambda functions

File: Tasks/plane.js [hunk 3]
```diff
@@ -1,19 +1,8 @@
...
+  const res = [];
+  for (const val of arr)
+    Array.isArray(val) ? res.push(...plane(val)) : res.push(val);
```
Comment (tshemsedinov) [inline Tasks/plane.js:6]: - Use if-statement instead of ternary here
- Add code block

File: Tasks/ip.js (hunk 1 above)
Comment (GhostDolphin) [reply Tasks/ip.js]: ip.js is fully fixed

File: Tasks/plane.js (hunk 2 above)
Comment (GhostDolphin) [reply Tasks/plane.js]: Fixed

File: Tasks/plane.js (hunk 3 above)
Comment (GhostDolphin) [reply Tasks/plane.js]: plane.js is fully fixed

File: Tasks/ip.js [hunk 4]
```diff
@@ -1,17 +1,9 @@
...
-  else {
-  B = i.split('.');
-  j = 0
-    for ( const b of B) {
-  a[j] = parseInt(b);
-  j++;
-  }};
-  return a;
+const parseIp = (ip) => {
+  if (ip !== '') {
```
Comment (tshemsedinov) [inline Tasks/ip.js:4]:
```suggestion
  if (ip === '') return null; // or undefined or null or even empty array []
```

**Pull request closed without merging**